src/browser/ui/ReactMount.js

```diff
@@ -517,6 +517,26 @@ var ReactMount = {
       )
     );
 
+    var containerOwnerDocument = null;
+    if (container) {
+      if (container.nodeType === DOC_NODE_TYPE) {
+        containerOwnerDocument = container;
+      }
+      if (container.nodeType === ELEMENT_NODE_TYPE ||
+        container.nodeType === DOCUMENT_FRAGMENT_NODE_TYPE) {
+        containerOwnerDocument = container.ownerDocument;
+      }
+    }
```

Comment:
We should put this whole block inside an `if (__DEV__)` block so that it's stripped out in prod.

The other thing we should consider is that we really shouldn't check this on every top level render. With the exception of iframes, once we confirm that document.body is there we don't ever need to do the expensive part of this (accessing the DOM).